src/addrman.h

```diff
@@ -631,9 +631,8 @@ class CAddrMan
     uint256 nKey;
 
     //! Source of random numbers for randomization in inner loops
-    FastRandomContext insecure_rand;
+    mutable FastRandomContext insecure_rand GUARDED_BY(cs);
 
-private:
```

Comment:
I don't really like changing all of addrman's private members to be protected. Since this is just to allow the tests to modify `nKey` and `insecure_rand`, what do you think about instead adding a `bool deterministic` argument to `CAddrMan`'s ctor (similar to the `deterministic` argument in `TxRequestTracker::TxRequestTracker`).

Reply:
Is there any risk allowing the unit tests access? Happy to add `private` back (after the Mutex) if you think it helps.

Reply:
Yeah, I'd prefer for `private` to be added after the mutex to minimize the change.

Reply:
Force pushed to add `private`